### ISSUE TITLE
Get stream from TV GO online live TV service

### ISSUE THREAD

**Issue by nyuszika7h** [streamlink/streamlink#784]
### Checklist

- [ ] This is a bug report.
- [ ] This is a plugin request.
- [x] This is a feature request.
- [x] I used the search function to find already opened/closed issues or pull requests.

### Description

I'm trying to capture a stream from my TV provider's online live TV service. However, it appears they are using a weak DH key for their HLS key file. I need to be able to ignore that error and proceed, because contacting them about this is unlikely to get it fixed soon. It plays fine in their own proprietary app. Unfortunately, I cannot provide a sample URL because it's a subscription service (although free for those with a TV subscription) and the URL contains account identifiers.

### Environment details (operating system, python version, etc.)

* Windows 10 x64
* Python 3.6.1
* streamlink 0.5.0

### Comments, logs, screenshots, etc.

```
[cli][info] Found matching plugin stream for URL hls://http://[REDACTED]
[cli][info] Available streams: live (best, worst)
[cli][info] Opening stream: live (hls)
[stream.hls][error] Failed to create decryptor: Unable to open URL: https://[REDACTED] ([SSL: SSL_NEGATIVE_LENGTH] dh key too small (_ssl.c:749))
[cli][error] No data returned from stream
[cli][info] Closing currently open stream...
```

**Comment by beardypig** [comment]
You can use the [--http-disable-dh](https://streamlink.github.io/cli.html#cmdoption-http-disable-dh) to disable Diffie-Hellman key exchange.

**Comment by nyuszika7h** [comment]
Oops, sorry. I found that in meantime. But now I get `[stream.hls][error] Failed to create decryptor: Unable to open URL: https://[REDACTED] ([SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:749))`. Let me try again, maybe the link expired.

**Comment by beardypig** [comment]
Try [--http-no-ssl-verify](https://streamlink.github.io/cli.html#cmdoption-http-no-ssl-verify) as well.

**Comment by nyuszika7h** [comment]
Still getting a handshake failure, not sure why.

**Comment by beardypig** [comment]
Does the key URL work in your browser?

**Comment by nyuszika7h** [comment]
Nope, I get the DH key too weak error and it doesn't let me proceed.

**Comment by beardypig** [comment]
Are you able to try with a command line tool like `curl`?

**Comment by nyuszika7h** [comment]
I tried, but I see no option in `curl` to disable this check. :/

**Comment by beardypig** [comment]
Try
```
curl --ciphers "DEFAULT:!DH" <url>
```

**Comment by nyuszika7h** [comment]
Now `curl` also gives me this error:

```
curl: (35) error:14094410:SSL routines:ssl3_read_bytes:sslv3 alert handshake failure
```

**Comment by beardypig** [comment]
You sure there isn't a client cert and key required? Probably it's included in their app, i assume it's an Android/iOS app.

**Comment by nyuszika7h** [comment]
I'm using their Windows app. I can get the play URL because that request is made by the browser, but the app refuses to play the stream if I put Fiddler decrypting SSL connections in the middle, even with Fiddler's cert trusted in the Windows certificate trust store.

There's also an Android app as well, but that doesn't work on rooted devices and while trying to set up Magisk which could hide root, I got my phone stuck in a bootloop, but that's off-topic here (don't worry, I know how to recover from it.)

**Comment by beardypig** [comment]
Probably it ignored the system level certs and has it's own built-in. I would download the apk for the Android app, extract/decompile it and look for a pem and crt file. If you want you can drop me an email, beardypig at protonmail dot com, with more details and I can take a look.

**Comment by nyuszika7h** [comment]
I edited before you responded – I found a cert and a key file, will try using those.

**Comment by nyuszika7h** [comment]
Now I'm getting another error:

```
[stream.hls][error] Failed to create decryptor: Unable to open URL: https://[REDACTED] ([SSL: TLSV1_ALERT_UNKNOWN_CA] tlsv1 alert unknown ca (_ssl.c:749))
```

**Comment by spunszor** [comment]
Any luck with solving this issue?

**Comment by back-to** [comment]
cable tv won't be supported by streamlink #1656

---

you can try the ssl ignore commands which you already did,
unlucky that it doesn't work

https://streamlink.github.io/cli.html#cmdoption-http-no-ssl-verify

but since this is not a streamlink issue I will close this.
